generator/ParamGenerator.cs

```diff
@@ -109,6 +110,18 @@ public void Execute(GeneratorExecutionContext context)
                             resetFields.AppendLine($"foreach (var item in {bbp.Name}) item?.Reset();");
                         }
                     }
+
+                    // Check to see if this is a Node[] or similar
+                    if (SymbolEqualityComparer.Default.Equals(namedType.ConstructedFrom, arrayType))
```

Comment:
Since both `generic type` and `named type` are the same underlying type we can probably abstract the if and the execution to its own method.

Reply:
Good catch! Updated.